main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopTimeIncreasingDistanceValidator.java

```diff
@@ -44,14 +44,29 @@ public class StopTimeIncreasingDistanceValidator extends FileValidator {
     this.stopTimeTable = stopTimeTable;
   }
 
+  @Override
+  public boolean shouldCallValidate() {
+    return stopTimeTable.hasColumn(GtfsStopTime.STOP_ID_FIELD_NAME)
+        && stopTimeTable.hasColumn(GtfsStopTime.SHAPE_DIST_TRAVELED_FIELD_NAME);
+  }
+
   @Override
   public void validate(NoticeContainer noticeContainer) {
     for (List<GtfsStopTime> stopTimeList : Multimaps.asMap(stopTimeTable.byTripIdMap()).values()) {
       // GtfsStopTime objects are sorted based on @SequenceKey annotation on stop_sequence field.
-      for (int i = 1; i < stopTimeList.size(); ++i) {
-        GtfsStopTime prev = stopTimeList.get(i - 1);
-        GtfsStopTime curr = stopTimeList.get(i);
-        if (prev.hasShapeDistTraveled()
+      GtfsStopTime prev = null;
+      GtfsStopTime curr;
+      for (int i = 0; i < stopTimeList.size(); ++i) {
```

Comment:
Nitpick: Now that it starts at 0 and i is not needed in the loop, `for (var curr : stopTimeList) {` would be enough.
Don't feel you have to change it.

Reply:
It's needed to set the `first` valid `prev` value needs to be set.

Reply:
Oh, I misread the feedback. Nice catch; yes, what you are proposing looks cleaner.